rfcs/011-resource-access-control/011-resource-access-control.md

```diff
@@ -6,136 +6,328 @@
+the above API through a user interface.  The interface would also surface errors reported from the API which verifies if
+the passed request is possible.
 
-### Isolating workloads in k8s
+### Advanced resource pools management
 
-Things are simplified by using a single namespace for all sessions. Resource quotas in k8s can use a `scopeSelector`
-to only work on specific priority classes (see [here](https://kubernetes.io/docs/concepts/policy/resource-quotas/#resource-quota-per-priorityclass). 
+In some cases, the administrator will need to dedicate specific Kubernetes nodes to resource pools, for instance when
+some user groups are allowed to use GPUs or when the autoscaling profiles for the nodes are different.
 
-Different modes of operation:
+In those cases, we rely on the `affinity` and `taint/toleration` mechanisms from Kubernetes. For each resource pool, the
+administrator will take care of:
+ - (optional) Setting a label on the nodes dedicated to the resource pool
+ - Setting a taint on the nodes dedicated to the resource pool
+ - Specify toleration and (optionally) a label when creating/updating the resource pool associated with the nodes
 
-1. No node taints, tolerations or affinities
+The above will result in three different modes of operation:
+
+1. No node taints/tolerations or affinities
   - sessions can be scheduled on any default or untainted session node (subject to resource availability and node sizes)
   - resource quotas still apply (based on priority class)
   - no guarantee which node exactly will be used for the session
-  - there is absolutely no connection between resource pools and k8s nodes
+  - there is no connection between resource pools and Kubernetes nodes
 
-2. Node taints and toleration, but no affinities
-  - sessions can be scheduled on the default pool, tained nodes or any other untainted nodes that "fit"
+2. Node taints/toleration, but no affinities
+  - sessions can be scheduled on the default pool, tainted nodes or any other untainted nodes that "fit"
   - resource quotas still apply (based on priority class)
   - no guarantee which node exactly will be used for the session
 
-3. Nodes taints, tolerations and affinities
+3. Preferred Nodes taints/tolerations and affinities
   - assuming affinities are `PreferredDuringSchedulingIgnoredDuringExecution`
-  - session will attempt to be scheduled on the specific nodes
+  - the session will attempt to be scheduled on the specific nodes
   - if specific nodes are full then the session can be scheduled anywhere else that fits
   - resource quotas still apply
 
-In almost all cases it can happen that a session is scheduled on an untainted node/resource pool that is not strictly
-related to the resource pool that the user was "targeting". In this case the request quota from the targeted resource
-pool is consumed.
-
-### Resource pool definition
-
-This is a YAML representation of the data that would be stored in postgres.
-
-```yaml
```

Comment:
I would like to keep some representation of the data that will be stored in postgres. Why do you want to fully remove this?

Reply:
I have added a section with a possible structure for the DB.